### PR TITLE
Step7

### PR DESCRIPTION
**Pull request by nitdevv**

### CODE REVIEW

File: src/app/edit.component/edit.component.ts [hunk 1]
```diff
@@ -1,54 +1,44 @@
...
 import * as _ from 'lodash';
+import 'rxjs/add/operator/switchMap';
+
 @Component({
-     selector: 'edit-component',
-    templateUrl: './edit.component.html',
-    styleUrls: ['../dashboard.component.css'],
-  })
+  selector: 'edit-component',
+  template: `<div *ngIf="todo">
```
Comment (Rahul-Khera-Codes) [inline src/app/edit.component/edit.component.ts]: use template url. not template directly in code

File: src/app/services/todo.service.ts [hunk 2]
```diff
@@ -3,14 +3,31 @@ import { Todo } from '../todo.data/todo';
...
 import * as _  from 'lodash';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/of';
+
 @Injectable()
 export class HeroService {
-    todos: Todo[] = TODOS;
-    getHeroes() {
-        return Promise.resolve(this.todos);
+    private _url: string = "http://localhost/angulardemo/src/data/tododata.json"
```
Comment (Rahul-Khera-Codes) [inline src/app/services/todo.service.ts]: make a constant file of url.

File: src/app/stats.component.ts [hunk 3]
```diff
@@ -10,13 +10,13 @@ import * as _  from 'lodash';
   <div class="container-fluid">
 
     <div class="navbar-header">
-        <button role="group" class="btn btn-default" type="button">
- 	  All <span class="badge"> {{done(todos)}}{{NoTodos}}</span>
+        <button role="group" class="btn btn-default" type="button" (click)="filterTodo('all')">
```
Comment (Rahul-Khera-Codes) [inline src/app/stats.component.ts]: why filter code in this  pull ? is was in your previous pull .

File: src/app/app.module.ts [hunk 4]
```diff
@@ -5,6 +5,9 @@ import { AppComponent } from '../app/app.component/app.component';
 import { DashboardComponent } from '../app/dashboard.component/todo.dashboard';
 import { EditComponent } from '../app/edit.component/edit.component';
 import { StatsComponent } from '../app/stats.component';
+import { AddComponent } from '../app/edit.component/add.component';
+import{HttpModule} from '@angular/http';
+import {HttpClientModule} from '@angular/common/http'
```
Comment (Rahul-Khera-Codes) [inline src/app/app.module.ts]: why import HttpClientModule ?

File: src/app/app.module.ts [hunk 5]
```diff
@@ -14,15 +17,20 @@ import{RouterModule} from '@angular/router';
...
     StatsComponent
+    AddComponent
   ],
   imports: [
     BrowserModule,
     RouterModule.forRoot([
       {path: '', pathMatch: 'full', redirectTo: 'dashboard'},
       {path:'dashboard',component:DashboardComponent},
       {path:'edit/:id',component:EditComponent},
+      {path:'add', component: AddComponent }
```
Comment (Rahul-Khera-Codes) [inline src/app/app.module.ts:27]: Add component in this pull ? it was in your previous pull. again ?

File: src/app/stats.component.ts [hunk 6]
```diff
@@ -10,13 +10,13 @@ import * as _  from 'lodash';
   <div class="container-fluid">
```
Comment (Rahul-Khera-Codes) [inline src/app/stats.component.ts]: use templateUrl not template.

Comment (nitdevv) [reply src/app/stats.component.ts]: ok sir

File: src/app/app.module.ts (hunk 5 above)
Comment (nitdevv) [reply src/app/app.module.ts:27]: in previous pull no seprate component made for add so i made the separte file and the their component

File: src/app/edit.component/edit.component.ts (hunk 1 above)
Comment (nitdevv) [reply src/app/edit.component/edit.component.ts]: ok sir

File: src/app/services/todo.service.ts (hunk 2 above)
Comment (nitdevv) [reply src/app/services/todo.service.ts]: ok sir

**Pull request merged**